src/monitor/process_monitor_main.cpp

```diff
@@ -53,6 +55,72 @@ namespace lo2s
 namespace monitor
 {
 
+static void drop_privileges()
+{
+    // get original uid & gid
+    gid_t original_uid;
+    uid_t original_gid;
+
+    if (config().user != "")
+    {
+        auto orig_user = getpwnam(config().user.c_str());
+        if (orig_user == nullptr)
+        {
+            Log::error() << "No user found with username '" + config().user + ".";
+            throw_errno();
+        }
+
+        original_uid = orig_user->pw_uid;
+        original_gid = orig_user->pw_gid;
+    }
+    else
+    {
+        // sudo is garanteed in this case, see check in config.hpp
+        try
+        {
+            original_uid = std::stoi(std::getenv("SUDO_UID"));
+            original_gid = std::stoi(std::getenv("SUDO_GID"));
+        }
+        catch (std::invalid_argument const& e)
+        {
+            Log::error() << "Cannot parse SUDO_UID/SUDO_GID into int.";
+            throw_errno();
+        }
+        catch (std::out_of_range const& e)
```

Comment:
I don't think the distinction between both exceptions types adds any value here. I'd rather have one "parsing failed" message.

Reply:
Would you remove the second catch statement and catch general exceptions or duplicate the error handling for both types? 
I'd go for the second option, but redundant code is wack and a function would be overkill.

Reply:
I'd go with one handler `catch(const std::exception& e)`. As users typically do not set these environment variables themselves, I don't think any other message than "failed to parse ..."  is necessary, but rather confusing.